### DISCUSSION TITLE
Evaluating model complexity in highs avoiding device related performance

### DISCUSSION THREAD

**Discussion by nunzioono** [ERGO-Code/HiGHS#1787] [Q&A] [answered]
Hi, i'm running multiple instances of a model with a parameter. I want to evaluate how does the model variates his complexity as i increase the above parameter. I tought to avoid performance related biases to evaluate the growing complexity as number of the iterations of the simplex within highs.
I wonder if does exist a way to get this info at the current version of highs or if is a valid suggestion for a feature.
Thanks for your work.

**Comment by jajhall** [comment]
Fundamentally, the simplex iteration count is available from the C++ data structure

https://ergo-code.github.io/HiGHS/dev/structures/classes/HighsInfo/#HighsInfo

The same information is available via other language interfaces. If not C++, what are you using?

**Reply by nunzioono** [reply]
Now i'm using the rust version of highs through the good_lp crate

**Reply by jajhall** [reply, marked as answer]
It would appear from

https://docs.rs/highs-sys/1.6.2/highs_sys/

that you can use

https://docs.rs/highs-sys/1.6.2/highs_sys/fn.Highs_getIntInfoValue.html

by passing the name "simplex_iteration_count"

**Reply by nunzioono** [reply]
Thanks

**Reply by nunzioono** [reply]
I'm sorry but looking around i couldn't find any example in rust. I wouldn't bother, but i wrote the code with the function to get the iteration number, but i get an access violation error (memory related) since probably the struct which i'm passing to the function is not the correct one.
Could you help me understand which struct sholud i pass?
This is the code:
`    let n = 4;

    let mut c = vec![(n+1)*(n+2)/2; n+1];

    c[4] = 12;

    let variables = vars(n);
    let mut problem = vars(n)
    .minimise(obj(n, &variables))
    .using(default_solver);

    for constraint in bounds(n, &c, &variables) {
        let _ = problem.add_constraint(constraint);
    }

    let now = std::time::Instant::now();
    let solution = problem.solve().unwrap();
    let solution_internal_ptr: *const c_void = &solution as *const _ as *const c_void;
    
    let mut iterations: i32 = 0;
    let iterations_ptr: *mut i32 = &mut iterations;

    unsafe {
        let info = CString::new("simplex_iteration_count").unwrap();
        let status = Highs_getIntInfoValue(solution_internal_ptr, info.as_ptr(), iterations_ptr);

        println!("Status: {:?}", status);
        println!("Iterations: {:?}", iterations);
    }

    let obj_expression = obj(n, &variables);
    if obj_value(solution, obj_expression) == ((n+1)*(n+2)/2) as f64 {
        println!("Solution is valid, solved in {:.2?}", now.elapsed());
             
    }`.
The struct i'm currently passing "solution_internal_ptr" in rust is an instance of HighsSolution

**Reply by nunzioono** [reply]
![image](https://github.com/ERGO-Code/HiGHS/assets/36959525/0384391e-5cee-4853-9b14-8214da740c78)

**Reply by jajhall** [reply]
Sorry, we didn't write the Rust interface, and don't know the language, so I can't help you. I suggest you contact @lovasoa and/or post an issue on the Rust repo

**Reply by nunzioono** [reply]
Ok thanks i'll try there

**Reply by lovasoa** [reply]
Hey ! Sorry to bounce, but that does not sound like a rust-related issue.

**Reply by nunzioono** [reply]
Thank you for the quick answer i wonder what the c_void* pointer should point to, i would bet my issue is there, i only found this example written in c on the [example](https://github.com/ERGO-Code/HiGHS/blob/13363c9f1252b015cf6527132eb9cf8f4b5bf020/src/interfaces/highs_csharp_api.cs#L382) but i didn't understand which struct in the highs rust api should resemble the problem instance, i tried with HighsProblem and HighsSolution but nothing

**Reply by nunzioono** [reply]
pub unsafe extern "C" fn Highs_getIntInfoValue(
    highs: [*const ](https://doc.rust-lang.org/nightly/std/primitive.pointer.html)[c_void](https://doc.rust-lang.org/nightly/std/os/raw/type.c_void.html),
    info: [*const ](https://doc.rust-lang.org/nightly/std/primitive.pointer.html)[c_char](https://doc.rust-lang.org/nightly/std/os/raw/type.c_char.html),
    value: [*mut ](https://doc.rust-lang.org/nightly/std/primitive.pointer.html)[HighsInt](https://docs.rs/highs-sys/1.6.2/highs_sys/type.HighsInt.html)
) -> [HighsInt](https://docs.rs/highs-sys/1.6.2/highs_sys/type.HighsInt.html)
